### ISSUE TITLE
Call to a member function getQueryBuilder() on string yiisoft/yii2/db/ActiveQuery.php:328

### ISSUE THREAD

**Issue by manoj-malviya** [yiisoft/yii2#20201]
We started facing this error when we upgraded from 2.0.48 to 2.0.50

### What steps will reproduce the problem?

We are passing slave DB connection in ActiveDataProvier

```
$dataProvider = new ActiveDataProvider(
                [
                    'query'      => $query,
                    'db'         => 'dbSlave',
                    'pagination' => ['pageSize' => 30],
                    'sort'       => false,
                ]
            );
```

Removing 'db' => 'dbSlave' from above line solves the issue.

Reverting to 2.0.48 also fixed the issue.

### What do you get instead?

Exception - Call to a member function getQueryBuilder() on string yiisoft/yii2/db/ActiveQuery.php:328

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.59
| PHP version      |  8.2.6
| Operating system | Centos 7

**Comment by vadimermolenko8787** [comment]
Have the same issue. It happens when u set in DataProvider pagination param, then latter Pagination objects created and after makes request to get total count, but property $db  of DataProvider is just "db"  instead of Connection object, that is why we got such error. As a quick win u could set db => null. 

![Снимок экрана 2024-06-12 в 11 46 09](https://github.com/yiisoft/yii2/assets/72613512/8179bbb2-27f9-4b7a-8b71-80d2ea12910a)

**Comment by samdark** [comment]
@lav45 would you please check it?

**Comment by lav45** [comment]
@manoj-malviya update yii2 to dev-master and check again
Your problem has already been solved in #20175

```json
 {
    "require": {
       "yiisoft/yii2": "dev-master"
   }
}
```
